sw/host/tests/manuf/manuf_cp_yield_test/src/main.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This test checks that we can change TAP strapping from CPU to the LC_CTRL
+//! and still use the interface without resetting.
+//!
+//! Both interfaces are used to check they're accessible. This is only possible
+//! in some lifecycle states: `TEST_UNLOCKED0` is used in this case.
+
+use anyhow::Context;
+use structopt::StructOpt;
+
+use opentitanlib::dif::lc_ctrl::{DifLcCtrlState, LcCtrlReg};
+use opentitanlib::io::jtag::{JtagParams, JtagTap};
+use opentitanlib::test_utils::extclk::{ClockSpeed, ExternalClock};
+use opentitanlib::test_utils::init::InitializeTest;
+use top_earlgrey::top_earlgrey_memory;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams,
+}
+
+fn main() -> anyhow::Result<()> {
+    let opts = Opts::from_args();
+    opts.init.init_target()?;
+
+    let transport = opts.init.init_target()?;
+
+    // Begin with the TAP straps set for the CPU.
+    transport
+        .pin_strapping("PINMUX_TAP_RISCV")?
+        .apply()
+        .context("failed to apply RISCV TAP strapping")?;
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset")?;
+
+    let jtag = transport.jtag(&opts.jtag)?;
+
+    // Connect to the CPU over JTAG.
+    jtag.connect(JtagTap::RiscvTap)
+        .context("failed to connect to RISCV TAP over JTAG")?;
+
+    ExternalClock::enable(&*jtag, ClockSpeed::High).context("failed to enable external clock")?;
+
+    // Read and write memory to verify connection.
+
+    // Check the first four words of the ROM.
+    let mut rom_read_buf = [0; 4];
+    jtag.read_memory32(
+        top_earlgrey_memory::TOP_EARLGREY_ROM_BASE_ADDR as u32,
+        &mut rom_read_buf,
+    )?;
+
+    let expected_rom_read = [0x4480006f, 0x4440006f, 0x4400006f, 0x43c0006f];
```

Comment:
Maybe to test the TAP, you should read the reset reasons from the retention SRAM (see [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_rom/test_rom.c;drc=2ecf53be82e9a8315416d0a4dc00314fbe3ddd58;l=162)) instead of reading the ROM, as any changes in the ROM would cause this test to fail. wdyt?

Reply:
Sure, I'm happy with that (and we already have `rstmgr` in bindgen)

Reply:
This register wasn't already in the bindgen setup and I was getting a weird value, so I've use the LC_CTRL's `LC_STATE` register for consistency with the other TAP read. Hope that's okay. This has a known value for this test, so shouldn't change.